purchases/src/main/kotlin/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -421,20 +417,29 @@ internal class BillingWrapper(
 
     internal fun acknowledge(
         token: String,
-        onAcknowledged: (billingResult: BillingResult, purchaseToken: String) -> Unit,
+        initiationSource: PostReceiptInitiationSource,
+        appInBackground: Boolean,
+        onAcknowledged: (purchaseToken: String) -> Unit,
     ) {
         log(LogIntent.PURCHASE, PurchaseStrings.ACKNOWLEDGING_PURCHASE.format(token))
-        executeRequestOnUIThread { connectionError ->
-            if (connectionError == null) {
-                withConnectedClient {
-                    acknowledgePurchase(
-                        AcknowledgePurchaseParams.newBuilder().setPurchaseToken(token).build(),
-                    ) { billingResult ->
-                        onAcknowledged(billingResult, token)
-                    }
-                }
-            }
-        }
+        AcknowledgePurchaseUseCase(
+            AcknowledgePurchaseUseCaseParams(
+                token,
+                initiationSource,
+                appInBackground,
+            ),
+            onReceive = onAcknowledged,
+            { error ->
+                // TODO-retry: if ITEM_NOT_OWNED queryPurchasesAsync
```

Comment:
oh right, forgot about this TODO. So Google recommends to call `queryPurchasesAsync` if consume or acknowledge returns `ITEM_NOT_OWNED`. I left this as a TODO because I think it's an edge case and need to think how to surface it. Not sure if the logic should be here in BillingWrapper or in the `Acknowledge`/`ConsumeBillingUseCase`. Wdyt?

Reply:
Hmm I think if it's part of the acknowledge process, we should try to put it inside that use case if needed... But yeah, I agree that seems like an edge case.

Reply:
yes... it's weird a usecase uses another usecase, but I did that in #1495 as well, so I think it's fine. That way that logic is encapsulated there